File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/Codes1.1/Codes1D/GradJacobiP.m
function [dP] = GradJacobiP(r, alpha, beta, N);

% function [dP] = GradJacobiP(r, alpha, beta, N);
% Purpose: Evaluate the derivative of the Jacobi polynomial of type (alpha,beta)>-1,
%	       at points r for order N and returns dP[1:length(r))]        

dP = zeros(length(r), 1);
if(N == 0)
  dP(:,:) = 0.0; 
else
  dP = sqrt(N*(N+alpha+beta+1))*JacobiP(r(:),alpha+1,beta+1, N-1);
end;
return

File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/nudg++/trunk/Src/Codes1D/GradJacobiP.cpp
// GradJacobiP.m
// function [dP] = GradJacobiP(z, alpha, beta, N);
// 2007/06/06
//---------------------------------------------------------
#include "NDGLib_headers.h"

//---------------------------------------------------------
DVec& GradJacobiP(const DVec& z,double alpha,double beta,int N)
//---------------------------------------------------------
{
  // function [dP] = GradJacobiP(z, alpha, beta, N);
  // Purpose: Evaluate the derivative of the orthonormal Jacobi
  //	   polynomial of type (alpha,beta)>-1, at points x
  //          for order N and returns dP[1:length(xp))]

  DVec* dP = new DVec(z.size(), 0.0, OBJ_temp, "dP");
  if (0 == N) {
    dP->fill(0.0);
  } else {
    (*dP) = sqrt(N*(N+alpha+beta+1))*JacobiP(z,alpha+1,beta+1, N-1);
  }
  return (*dP);
}
